packages/okta-auth-js/lib/browser/browser.js

```diff
@@ -219,105 +218,104 @@ proto.signIn = function (opts) {
+    if (e.name === 'AuthApiError') {
+      // Capture and ignore network errors
+      return;
+    }
+    throw e;
+  });
+};
+
+// Revokes accessToken, clears all local tokens, then redirects to Okta to end the SSO session.
+proto.signOut = async function (options) {
   options = util.extend({}, options);
 
   // postLogoutRedirectUri must be whitelisted in Okta Admin UI
-  var postLogoutRedirectUri = options.postLogoutRedirectUri || this.options.postLogoutRedirectUri;
+  var defaultUri = window.location.origin;
+  var postLogoutRedirectUri = options.postLogoutRedirectUri
+    || this.options.postLogoutRedirectUri
+    || defaultUri;
 
   var accessToken = options.accessToken;
-  var revokeAccessToken = options.revokeAccessToken;
+  var revokeAccessToken = options.revokeAccessToken !== false;
   var idToken = options.idToken;
 
   var sdk = this;
   var logoutUrl = oauthUtil.getOAuthUrls(sdk).logoutUrl;
 
-  function getAccessToken() {
-    return Promise.resolve()
-    .then(function() {
-      if (revokeAccessToken && typeof accessToken === 'undefined') {
-        return sdk.tokenManager.get('token');
-      }
-      return accessToken;
-    });
+  if (typeof idToken === 'undefined') {
+    idToken = await sdk.tokenManager.get('idToken');
   }
 
-  function getIdToken() {
-    return Promise.resolve()
-    .then(function() {
-      if (postLogoutRedirectUri && typeof idToken === 'undefined') {
-        return sdk.tokenManager.get('idToken');
-      }
-      return idToken;
-    });
+  if (revokeAccessToken && typeof accessToken === 'undefined') {
+    accessToken = await sdk.tokenManager.get('token');
   }
 
-  function closeSession() {
-    return sdk.session.close() // DELETE /api/v1/sessions/me
-    .catch(function(e) {
-      if (e.name === 'AuthApiError') {
-        // Most likely cause is session does not exist or has already been closed
-        // Could also be a network error. Nothing we can do here.
-        return;
-      }
-      throw e;
-    });
+  // Clear all local tokens
+  sdk.tokenManager.clear();
+
+  if (revokeAccessToken && accessToken) {
+    await sdk.revokeAccessToken(accessToken);
   }
 
-  return allSettled([getAccessToken(), getIdToken()])
-    .then(function(tokens) {
-      accessToken = tokens[0].value;
-      idToken = tokens[1].value;
-
-      // Clear all local tokens
-      sdk.tokenManager.clear();
-
-      if (revokeAccessToken && accessToken) {
-        return sdk.token.revoke(accessToken)
-        .catch(function(e) {
-          if (e.name === 'AuthApiError') {
-            // Capture and ignore network errors
-            return;
-          }
-          throw e;
-        });
-      }
+  // No idToken? This can happen if the storage was cleared.
+  // Fallback to XHR signOut, then redirect to the post logout uri
+  if (!idToken) {
+    return sdk.closeSession()
+    .catch(function(err) {
+      // eslint-disable-next-line no-console
+      console.log('Unhandled exception while closing session', err);
```

Comment:
Is there anything they can do to handle such exceptions?  If so we should mention what.  If not, this comment is a waste of your time and why are you still reading?

Reply:
We are catching and ignoring `AuthApiError` which would occur if the session has already been closed. This was the most common error. At this time I can't name a specific error we might be expecting, I'm leaning towards removing the `catch()` block from the documentation

Reply:
I am changing this to only capture the specific error related to session not found. All other errors will throw

Reply:
One possible error is 429 TOO_MANY_REQUESTS_EXCEPTION

Can they do anything about it? Not really, I am including the log so that it may be included with any bug or issue reports.

If we throw the error then the final step of changing window.location will not happen. We are already in an error fallback state here because the idToken could not be found.

The user's expectation is that this method should sign the user out and perform a redirect to Okta, therefore any exception thrown from this method is problematic. If the `catch()` block is reached it means the redirect did NOT occur, and also the user is probably not signed out.  Eating the exception here doesn't really mitigate that. If there is an exception from `closeSession` it probably means the user has not been signed out. So I think not eating the error and instead throwing the exception up would be a better choice.